blockchain/state/doc.go

```diff
@@ -18,5 +18,26 @@
 // This file is derived from core/state/statdb.go (2018/06/04).
 // Modified and improved for the klaytn development.
 
-// Package state provides a caching layer atop the Klaytn state trie.
+/*
+Package state provides a caching layer atop the Klaytn state trie.
+This package is used to read/write stateObject from/to StateDB and it also acts as a bridge between
+the objects and the persistent layer.
+
+Overview of state package
+
+stateObject represents a Klaytn account which is being modified. It is identified by its address.
+Once it is loaded from the persistent layer, it is cached and managed by StateDB.
+StateDB manages stateObjects and operations to them. Most of the operations are done by StateDB,
```

Comment:
```suggestion
StateDB stores and returns stateObjects. Most of the retrievals are done by StateDB,
```

Reply:
@KimKyungup StateDB does not just store and return stateObject. operations != retrievals, actual operations are mediated by StateDB, not direct access to the StateDB.

Reply:
I like the comment in `statedb.go`. How about using the expression? 
By the way, we should change 'ethereum protocol' to 'Klaytn protocol'.
```
StateDBs within the ethereum protocol are used to store anything
within the merkle trie. StateDBs take care of caching and storing
nested states.
```

Reply:
@aidan-kwon Thanks for the comment but "used to store anything within the merkle trie" is not what StateDB exactly does. Only a portion of merkle trie's leaf nodes are cached. Also I didn't understand what "nested states" means here. However, let me change the expression here.

Reply:
@ehnuje Thank you for the consideration. 
Could you change the comment also? Since we modified some part of Ethereum's StateDB, the description needs to be changed little bit. I think you can give a better description of StateDB than me. 
https://github.com/klaytn/klaytn/blob/9e2eb69ababca1d2f69f7f1b969480466cadddbb/blockchain/state/statedb.go#L58-L60

Reply:
@aidan-kwon Updated!

Reply:
Thank you @ehnuje !